src/main/java/org/opensearch/security/securityconf/IndexPattern.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.securityconf;
+
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.security.support.WildcardMatcher;
+import org.opensearch.security.user.User;
+
+import java.util.List;
+import java.util.Set;
+
+public interface IndexPattern {
+    IndexPattern addFlsFields(List<String> flsFields);
+
+    IndexPattern addMaskedFields(List<String> maskedFields);
+
+    WildcardMatcher getPerms();
+
+    Set<String> getStringPerm();
+
+    WildcardMatcher getNonWildCardPerms();
+
+    String getDlsQuery(User user);
+
+    Set<String> concreteIndexNames(User user, IndexNameExpressionResolver resolver, ClusterService cs);
```

Comment:
Nit: Can we rename this? I am not sure what a "concrete" index name means. I assume that it is the full name?

Reply:
This change has been reverted